### PR TITLE
CIF-884 - Prepare CIF Archetype Release

### PR DESCRIPTION
**Pull request by mhaack**
## Description

This PR adds the configuration needed to release the CIF Archetype via maven and CircleCI. It follows a similar approach like CIF Core Components release, where we prepare the release locally on a dev machine using `mvn release:prepare release:clean`, review it and push it to GitHub. CircleCI then picks up and performs the release to Maven Central.

Right now the config is prepared to follow a similar pattern like our other projects:
* Release tag must be pushed manually after reviewing the changes
* Maven Central upload is not published automatically (for this first release) to have the chance to review here as well before making public

## Related Issue

CIF-884

## Motivation and Context

Prepare everything needed to create the first release of this project.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,25 @@
...
+| autoInstallBundle         | Install core bundle with the maven-sling-plugin to the Felix console                                                                                                                                                                                          |
+| autoInstallPackage        | Install the ui.content and ui.apps content package with the content-package-maven-plugin to the package manager to default author instance on localhost, port 4502. Hostname and port can be changed with the aem.host and aem.port user-defined properties.  |
+| autoInstallPackagePublish | Install the ui.content and ui.apps content package with the content-package-maven-plugin to the package manager to default publish instance on localhost, port 4503. Hostname and port can be changed with the aem.host and aem.port user-defined properties. |
 
 ## Usage
+
 To use a released version of this archetype use the following Maven command:
+
 ```bash
 mvn archetype:generate \
```
Comment (herzog31) [inline README.md:24]: Please update the version to `x.y.z` in this command as well.

**Review by herzog31 — APPROVED** [review]

**Review by cjelger — COMMENTED** [review]
My only comment is about the versioning scheme, it seems other AEM archetypes simply use a major version without minor nor path. Do we want to also align on that?

**Comment by mhaack** [comment]
@cjelger I thought about that as well and we can do. I had a quick chat with @herzog31 and changed it to align with the CIF component versioning. Assuming we will release new CIF core components and always updated the archetype ... with major version only this will end up in >10 version quickly. WDYT?

**Comment by cjelger** [comment]
Sure, aligning with the components version makes sense, fine for me.

**Review by LSantha — APPROVED** [review]

**Pull request merged**